modules/images-triggering-updates-imagestream-changes-kubernetes.adoc

```diff
@@ -0,0 +1,67 @@
+// Module included in the following assemblies:
+//
+// * openshift_images/triggering-updates-on-imagestream-changes.adoc
+
+
+[id="images-triggering-updates-imagestream-changes-kubernetes_{context}"]
+= Triggering Kubernetes resources
+
+Kubernetes resources do not have fields for triggering, unlike deployment and build configurations, which include as part of their API definition a set of fields for controlling triggers. Instead, you can use annotations in {product-title} request triggering.
+
+The annotation is defined as follows:
+
+[source,yaml]
+----
+Key: image.openshift.io/triggers
+Value: <1>
+[
+ {
+   "from" :{
+     "kind": "ImageStreamTag", <2>
+     "name": "example:latest", <3>
+     "namespace": "myapp", <4>
+   },
+   "fieldPath": "spec.template.spec.containers[?(@.name='web')].image", <5>
+   "paused": "false" <6>
+ },
+ ...
+]
+----
+<1> `Value` is the array of triggers, where each item has the `schema:`.
+<2> `kind` is the resource to trigger from must be `ImageStreamTag`. This field is required.
+<3> `name` must be the name of an image stream tag. This field is required.
+<4> `namespace` defaults to the namespace of the object. This field is optional.
+<5> `fieldpath` is the JSON path to change. This field is limited, and only accepts a very specific set of inputs, a JSON path expression that precisely matches a container by ID or index. For pods the JSON path would be "spec.containers[?(@.name='web')].image". This field is required.
+<6>  `paused` defaults to `false`. Set to true to temporarily disable this trigger. This field is optional
+
+When {product-title} sees one of the core Kubernetes resources that contains both a pod template and this annotation, it attempts to update the object using the image currently associated with the image stream tag referenced by trigger. The update is performed against the `fieldPath` specified.
+
+Examples of core Kubernetes resources that can contain both a pod template and annotation include:
+
+* `CronJobs`
+* Deployments
+* `StatefulSets`
+* `DaemonSets`
+* Jobs
+* `ReplicationControllers`
+* Pods
+
+.Procedure
+
+* Trigger Kubernetes resources by updating the image stream tag.
+** In the following example the trigger fires when the `example:latest` image stream tag is updated. Upon firing, the pod template image reference for the `web` container is updated with a new image value for the object. If the pod template is part of a deployment definition, the change to the pod template automatically triggers a deployment, effectively rolling out the new image.
++
+[source,yaml]
+----
+image.openshift.io/triggers=[{"from":{"kind":"ImageStreamTag","name":"example:latest"},"fieldPath":"spec.template.spec.containers[?(@.name='web')].image"}]
+----
+
+* Trigger Kubernetes resources by entering the `oc set triggers` command.
```

Comment:
I think that these two options should be better fleshed out and presented in separate modules. They aren't sequential steps, so I don't think they belong in the same module.